From cfd5636e840e6f1fe92f70709a8638c522d4e8f0 Mon Sep 17 00:00:00 2001
From: Peter Chen <peter.chen@nxp.com>
Date: Wed, 16 Nov 2016 09:20:58 +0800
Subject: [PATCH 1338/5242] MLK-13570-1 ARM: dts: imx7ulp-evk: add USB support

commit  e26d6439ce78f2b64fcec6475f847e6bdf683c16 from
https://source.codeaurora.org/external/imx/linux-imx.git

Add USBOTG1 support, we use GPIO as ID function for dual-role switch.
Besides, #define <dt-bindings/gpio/gpio.h> to imx7ulp.dtsi since
lots of boards may need it.

Signed-off-by: Peter Chen <peter.chen@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/boot/dts/imx7ulp-evk.dts |   41 ++++++++++++++++++++++++++++++++++++-
 arch/arm/boot/dts/imx7ulp.dtsi    |    1 +
 2 files changed, 41 insertions(+), 1 deletion(-)

diff --git a/arch/arm/boot/dts/imx7ulp-evk.dts b/arch/arm/boot/dts/imx7ulp-evk.dts
index e7e6844..81c6860 100644
--- a/arch/arm/boot/dts/imx7ulp-evk.dts
+++ b/arch/arm/boot/dts/imx7ulp-evk.dts
@@ -9,7 +9,6 @@
 /dts-v1/;
 
 #include "imx7ulp.dtsi"
-#include <dt-bindings/gpio/gpio.h>
 
 / {
 	model = "NXP i.MX7ULP EVK";
@@ -46,6 +45,25 @@
 			startup-delay-us = <100>;
 			enable-active-high;
 		};
+
+		reg_usb_otg1_vbus: regulator@0 {
+			compatible = "regulator-fixed";
+			reg = <0>;
+			pinctrl-names = "default";
+			pinctrl-0 = <&pinctrl_usb_otg1>;
+			regulator-name = "usb_otg1_vbus";
+			regulator-min-microvolt = <5000000>;
+			regulator-max-microvolt = <5000000>;
+			gpio = <&gpio0 0 GPIO_ACTIVE_HIGH>;
+			enable-active-high;
+		};
+	};
+
+	extcon_usb1: extcon_usb1 {
+		compatible = "linux,extcon-usb-gpio";
+		id-gpio = <&gpio0 8 GPIO_ACTIVE_HIGH>;
+		pinctrl-names = "default";
+		pinctrl-0 = <&pinctrl_extcon_usb1>;
 	};
 
 	pf1550-rpmsg {
@@ -183,6 +201,18 @@
 			>;
 		};
 
+		pinctrl_usb_otg1: usbotg1grp {
+			fsl,pins = <
+				IMX7ULP_PAD_PTC0__PTC0		0x30100
+			>;
+		};
+
+		pinctrl_extcon_usb1: extcon1grp {
+			fsl,pins = <
+				IMX7ULP_PAD_PTC8__PTC8		0x30103
+			>;
+		};
+
 		pinctrl_usdhc1: usdhc1grp {
 			fsl,pins = <
 				IMX7ULP_PAD_PTE3__SDHC1_CMD	0x843
@@ -284,6 +314,15 @@
 	status = "okay";
 };
 
+&usbotg1 {
+	vbus-supply = <&reg_usb_otg1_vbus>;
+	extcon = <0>, <&extcon_usb1>;
+	srp-disable;
+	hnp-disable;
+	adp-disable;
+	status = "okay";
+};
+
 &usdhc0 {
 	pinctrl-names = "default", "sleep";
 	pinctrl-0 = <&pinctrl_usdhc0>;
diff --git a/arch/arm/boot/dts/imx7ulp.dtsi b/arch/arm/boot/dts/imx7ulp.dtsi
index cc22969..9b9247d 100644
--- a/arch/arm/boot/dts/imx7ulp.dtsi
+++ b/arch/arm/boot/dts/imx7ulp.dtsi
@@ -8,6 +8,7 @@
 
 #include <dt-bindings/clock/imx7ulp-clock.h>
 #include <dt-bindings/interrupt-controller/arm-gic.h>
+#include <dt-bindings/gpio/gpio.h>
 #include "skeleton.dtsi"
 #include "imx7ulp-pinfunc.h"
 
-- 
1.7.9.5

